react-native-ts/src/OfflineModeButton.tsx

```diff
@@ -0,0 +1,45 @@
+import React, {useEffect} from 'react';
+import {Pressable, StyleSheet} from 'react-native';
+import {Icon} from 'react-native-elements';
+import RealmContext from './RealmContext';
+
+const {useRealm} = RealmContext;
+
+export function OfflineModeButton() {
+  const realm = useRealm();
+
+  const [offlineMode, setOfflineMode] = React.useState(
+    realm.syncSession?.state === 'inactive',
+  );
+  // The signOut function calls the logOut function on the currently
+  // logged in user and then navigates to the welcome screen
+
+  useEffect(() => {
+    if (realm.syncSession?.state === 'active' && offlineMode === true) {
+      realm.syncSession.pause();
+    } else if (
+      realm.syncSession?.state === 'inactive' &&
+      offlineMode === false
+    ) {
+      realm.syncSession.resume();
+    }
```

Comment:
[idea] In the edge case, we might want `else setOfflineMode(!offlineMode)` (or log a warning to console?) -- then at least offlineMode will definitely match syncSession state.

Does SyncSession.addConnectionNotification() let us know when pause/resume happened? If so, maybe we could wire up a version that doesn't even use a separate state variable (to avoid ever getting in a weird state where the icon doesn't match the actual sync session state)

Reply:
I could make another `useEffect` to capture this case.

Reply:
I actually was able to remove `useState` using this method.  However, I did have to incorporate a force re-render, so that when the sync state changes, the icon re-renders.

Reply:
Nice! I think that works fine. Potentially you could stash the reducer 'hack' in a custom `useRerender()` hook that just gives you the `rerender` function back.